diff --git a/arch/arm/mach-tegra/sleep.S b/arch/arm/mach-tegra/sleep.S
index 18b8799..35ec19c 100644
--- a/arch/arm/mach-tegra/sleep.S
+++ b/arch/arm/mach-tegra/sleep.S
@@ -232,6 +232,7 @@ ENDPROC(tegra_cpu_pllp)
 #define TEGRA_SMC_SAVED_WORDS   11
 
 /* SMC issued using the current cacheable SP stack */
 ENTRY(tegra_generic_smc)
+	.arch_extension sec
 	mov     r3, sp                                  @ use current stack
        sub     r3, #(TEGRA_SMC_SAVED_WORDS << 2)       @ context grows up
        b       tegra_generic_smc_cached
